choreography/src/choreography/Fabrikam.Choreography.ChoreographyService/Controllers/ChoreographyController.cs

```diff
@@ -10,8 +10,8 @@
 using Fabrikam.Choreography.ChoreographyService.Services;
 using Fabrikam.Communicator.Service.Operations;
 using Microsoft.AspNetCore.Mvc;
-using Microsoft.Azure.EventGrid;
-using Microsoft.Azure.EventGrid.Models;
+using Azure.Messaging.EventGrid;
```

Comment:
Truthfully, though we didn't change the behavior with the upgrade, we should really look at this controller in more depth.   The logic is convoluted and it's demonstrating a really bad pattern of HTTP responses where any downstream failures are returned to the caller as an HTTP 400 (Bad Request) indicating "you did something wrong when sending your payload and you need to fix it for this to work."

Reply:
The coding conventions also often cut against best practices for C#.   For example:

```csharp
try
{
    ...
}
catch (Exception ex) when (ex is SomeExceptionType)
{
    ...
}
```

Should just be:

```csharp
try
{
    ...
}
catch (SomeExceptionType ex)
{
    ...
}
```

Reply:
We have known the existence of this problem, we will improve the logic after completing all migrations. Our primary task is upgrading sdk from t1 to t2.

Reply:
I realize this request expands the scope, but let's get a better design here as well.  Do you want to take a stab at it with Jesse's recommendation above or do you need further guidance?

Reply:
We will take a stab at it, but it maybe takes a few time.

Reply:
Did you resolve this?

Reply:
@jongio,@jsquire We tried to make some changes in exception handing. We separated some exceptions, but we can't break down the BadRequest of 400.